examples/flutter_gallery/lib/demo/pesto_demo.dart

```diff
@@ -0,0 +1,683 @@
+                    fit: ImageFit.none
+                  )
+                )
+              );
+            }
+          )
+        ),
+        drawer: new Drawer(
+          child: new Block(children: <Widget>[
+            new DrawerHeader(
+              content: new Column(
+                children: <Widget>[
+                  new Container(
+                    decoration: new BoxDecoration(
+                      border: new Border.all(color: _kTheme.primaryColor, width: 2.0),
+                      shape: BoxShape.circle
+                    ),
+                    width: 72.0,
+                    height: 72.0,
+                    padding: const EdgeInsets.all(2.0),
+                    margin: const EdgeInsets.only(bottom: 8.0),
+                    child: new ClipOval(
+                      child: new AssetImage(
+                        name: _kUserImage,
+                        fit: ImageFit.contain
+                      )
+                    )
+                  ),
+                  new Text(_kUserName, style: userStyle),
+                  new Text(_kUserEmail, style: emailStyle)
+                ]
+              )
+            ),
+            new DrawerItem(
+              child: new Text('Home'),
+              selected: !config.showFavorites,
+              onPressed: () => Navigator.pushNamed(context, PestoDemo.routeName)
+            ),
+            new DrawerItem(
+              child: new Text('Favorites'),
+              selected: config.showFavorites,
+              onPressed: () { _showFavorites(context); }
+            )
+          ])
+        ),
+        floatingActionButton: new FloatingActionButton(
+          child: new Icon(icon: Icons.edit),
+          onPressed: () { }
+        ),
+        body: new Block(
+          padding: new EdgeInsets.only(top: _kAppBarHeight + statusBarHeight),
+          children: <Widget>[
+            new Padding(
+              padding: const EdgeInsets.all(8.0),
+              child: new MaxTileWidthGrid(
+                maxTileWidth: 500.0,
+                children: recipes.map(
+                  (Recipe r) => _buildCard(context, r)
+                ).toList()
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+
+  Widget _buildCard(BuildContext context, Recipe recipe) {
+    final TextStyle titleStyle = _textStyle(24.0, FontWeight.w600);
+    final TextStyle authorStyle = _textStyle(12.0, FontWeight.w500).copyWith(color: Colors.black54);
+
+    return new Padding(
+      padding: const EdgeInsets.all(4.0),
+      child: new GestureDetector(
+        onTap: () => _showRecipe(context, recipe),
+        child: new Card(
+          child: new Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: <Widget>[
+              new AssetImage(
+                name: recipe.imagePath,
+                fit: ImageFit.scaleDown
+              ),
+              new Flexible(child:
```

Comment:
I suppose this sort of thing is going to come up a bunch here. We don't usually employ this line-saving shorthand.